projects/RabbitMQ.Client/client/api/ConnectionConfig.cs

```diff
@@ -46,23 +47,27 @@ public sealed class ConnectionConfig
         /// Virtual host to access during this connection.
         /// </summary>
         public string VirtualHost { get; }
+
         /// <summary>
         /// Username to use when authenticating to the server.
         /// </summary>
         public string UserName { get; }
+
         /// <summary>
         /// Password to use when authenticating to the server.
         /// </summary>
         public string Password { get; internal set; }
+
         /// <summary>
         ///  SASL auth mechanisms to use.
         /// </summary>
-        public IList<IAuthMechanismFactory> AuthMechanisms { get; }
+        public IEnumerable<IAuthMechanismFactory> AuthMechanisms { get; }
```

Comment:
This should be a IReadOnlyList<IAuthMechanismFactory>

Reply:
IEnumerable is the minimum interface necessary, and is immutable.

Reply:
And it makes it necessary to allocate when you iterate over it, which is not the case with the IReadOnlyList.

Reply:
You chose a IReadOnlyDictionary for the ClientProperties, but the minimum interface would be IEnumerable<KeyValuePair<string, object?>>. You didn't choose it for a reason, why is this one here any different?

Reply:
> IEnumerable<KeyValuePair<string, object?>>. You didn't choose it for a reason, why is this one here any different?

Because I didn't think of it. I will change that as well.

Reply:
I didn't think of that! No matter I'm planning to revert these changes anyway since `ConnectionConfig` shouldn't be modified by users.

I'm just used to Erlang where things are actually immutable by default.

Reply:
Interesting, I think I need to read up on Erlang =)